### PR TITLE
Serverpool in ZK

### PR DESCRIPTION
**Pull request by freva**

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java [hunk 1]
```diff
@@ -147,6 +147,14 @@ public Lock lockProvisionState(String provisionStateId) {
         return lock(lockPath(provisionStateId), Duration.ofMinutes(30));
     }
 
+    public Lock lockVespaServerPool() {
+        return lock(root.append("locks").append("vespaServerPoolLock"), defaultLockTimeout);
```
Comment (andreer) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java]: "lock" is already in the path so no need for it in the node name imho

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java [hunk 2]
```diff
@@ -147,6 +147,14 @@ public Lock lockProvisionState(String provisionStateId) {
...
+    }
+
+    public Lock lockOpenStackServerPool() {
+        return lock(root.append("locks").append("openStackServerPoolLock"), defaultLockTimeout);
```
Comment (andreer) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java]: likewise

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java (hunk 1 above)
Comment (andreer) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java]: Also, updating the server pool should never take more than a few seconds at most, as it's always just writing data to ZK - so we could probably use a much shorter lock timeout

**Review by andreer — APPROVED** [review]

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java (hunk 1 above)
Comment (freva) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java]: Guess it makes sense to have short timeout, but I'll keep path the same so at least its consistent with the already existing locks.

**Pull request merged**